Apps/WebClient/src/NewClientApp/src/components/navmenu/SidebarComponent.vue

```diff
@@ -0,0 +1,187 @@
+<script setup lang="ts">
+import { computed, ref } from "vue";
+
+import { useNavbarStore } from "@/stores/navbar";
+import { useUserStore } from "@/stores/user";
+import { useAuthStore } from "@/stores/auth";
+import { useConfigStore } from "@/stores/config";
+import { useRoute } from "vue-router";
+import HgIconButtonComponent from "@/components/shared/HgIconButtonComponent.vue";
+import { OidcUserInfo } from "@/models/user";
+import { useAppStore } from "@/stores/app";
+
+const route = useRoute();
+
+const appStore = useAppStore();
+const authStore = useAuthStore();
+const configStore = useConfigStore();
+const navbarStore = useNavbarStore();
+const userStore = useUserStore();
+
+const collapsedOnDesktop = ref(false);
+
+const isMobile = computed(() => appStore.isMobile);
+const isSidebarOpen = computed(() => navbarStore.isSidebarOpen);
+const isDependentEnabled = computed(
+    () => configStore.webConfig.featureToggleConfiguration.dependents.enabled
+);
+const isServicesEnabled = computed(
+    () => configStore.webConfig.featureToggleConfiguration.services.enabled
+);
+const oidcUserInfo = computed<OidcUserInfo | undefined>(
+    () => userStore.oidcUserInfo
+);
+
+const isQueuePage = computed(
+    () =>
+        route.path.toLowerCase() === "/queue" ||
+        route.path.toLowerCase() === "/busy"
+);
+
+const isSidebarAvailable = computed(
+    () =>
+        !configStore.isOffline &&
+        authStore.oidcIsAuthenticated &&
+        userStore.isValidIdentityProvider &&
+        userStore.userIsRegistered &&
+        userStore.userIsActive &&
+        !userStore.patientRetrievalFailed &&
+        !isQueuePage.value
+);
+
+const userInitials = computed<string>(() => {
+    const first = oidcUserInfo.value?.given_name;
+    const last = oidcUserInfo.value?.family_name;
+    if (first && last) {
+        return first.charAt(0) + last.charAt(0);
+    } else if (first) {
+        return first.charAt(0);
+    } else if (last) {
+        return last.charAt(0);
+    } else {
+        return "?";
+    }
+});
+
+const userName = computed<string>(() =>
+    oidcUserInfo.value === undefined
+        ? ""
+        : `${oidcUserInfo.value.given_name} ${oidcUserInfo.value.family_name}`
+);
```

Comment:
I think it would be worthwhile to move these into the user store so they can be accessed across the application.

Reply:
Changed.